components/builder-api-client/src/lib.rs

```diff
@@ -148,16 +151,34 @@ impl Client {
     /// # Failures
     ///
     /// * Remote API Server is not available
-    pub fn job_group_promote(&self, group_id: u64, channel: &str, token: &str) -> Result<()> {
+    pub fn job_group_promote(
+        &self,
+        group_id: u64,
+        channel: &str,
+        token: &str,
+    ) -> Result<Vec<String>> {
         debug!("Promoting job group {} to channel {}", group_id, channel);
         let url = format!("jobs/group/{}/promote/{}", group_id, channel);
         let res = self.add_authz(self.0.post(&url), token).send().map_err(
             Error::HyperError,
         )?;
+
         if res.status != StatusCode::Ok {
             return Err(err_from_response(res));
         }
-        Ok(())
+
+        match decoded_response::<UnsuccessfulJobGroupPromote>(res)
+            .map_err(Error::HabitatHttpClient) {
+            Ok(value) => Ok(value.projects),
+            Err(Error::HabitatHttpClient(hab_http::Error::Json(e))) => {
+                if e.is_eof() {
+                    return Ok(Vec::new());
```

Comment:
This error is specific to the way that serde returns things when it parses JSON that is an empty string, e.g. the server returned an empty response body.  The server is returning either nothing, indicating success, or a list of projects that failed to promote.  I guess I could change the server to return an empty list of projects that failed to promote, but imo, that makes the server API more awkward.  If a request succeeds, we don't return an empty list of errors - we just return nothing.  Either way, something is a little awkward, and imo, this strikes the right balance as it stands.